error: 'string_int' undeclared

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.12 9381714, press to see full `v doctor` output</summary>

|V full version      |V 0.4.12 9381714b5897072238cd3ccbf9822c6e2ba61fde
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 24.04.2 LTS
|Processor           |8 cpus, 64bit, little endian, 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
|Memory              |0.62GB/7.56GB
|                    |
|V executable        |/opt/v/v
|V last modified time|2025-12-18 10:25:35
|                    |
|V home dir          |OK, value: /opt/v
|VMODULES            |OK, value: /home/arghyas/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/arghyas/Programs/v_playground
|                    |
|Git version         |git version 2.43.0
|V git status        |N/A
|.git/config present |false
|                    |
|cc version          |cc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|gcc version         |gcc (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
|clang version       |N/A
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.39-0ubuntu8.6) 2.39

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  calc.v && calc`
```v
import os


fn main() {
	if os.args.len < 2 {
		usage()
		panic("Invalid Invocation")
	}

	option := os.args[1]
	arg1 := os.args[2]
	arg2 := os.args[3]

	result := match option {
		'-a', '--add' { add(arg1.int, arg2.int) }
		'-s', '--sub' { sub(arg1.int, arg2.int) }
		else { panic("Invalid Option") }
	}

	println(result)
}

fn usage() {
	println("BASIC CALCULATOR")
	println("Usage:")
	println("\tcalc --add 2 3")
	println("\tcalc --sub 5 2")
}

fn add(x int, y int) int {
	return x+y
}

fn sub(x int, y int) int {
	return x-y
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/calc.01KCRJ6FM9CT1E575CYX1WNFNH.tmp.c:6097: error: 'string_int' undeclared
=================================================================
Try passing `-g` when compiling, to see a .v file:line information, that correlates more with the C error.
(Alternatively, pass `-show-c-output`, to print the full C error message).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

Compiling and running this program without passing any arguments(`v run calc.v`) the output should have been "BASIC CALCULATOR"

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1900,8 +1900,10 @@ fn (mut c Checker) selector_expr(mut node ast.SelectorExpr) ast.Type {
 	if mut method := c.table.sym(c.unwrap_generic(typ)).find_method_with_generic_parent(field_name) {
 		c.markused_comptime_call(typ.has_flag(.generic), '${int(method.params[0].typ)}.${field_name}')
 		if c.expected_type != 0 && c.expected_type != ast.none_type {
-			fn_type := ast.new_type(c.table.find_or_register_fn_type(method, false, true))
-			// if the expected type includes the receiver, don't hide it behind a closure
+			mut method_copy := method
+			method_copy.name = ''
+			fn_type := ast.new_type(c.table.find_or_register_fn_type(method_copy, false,
+				true))
 			if c.check_types(fn_type, c.expected_type) {
 				c.table.used_features.anon_fn = true
 				return fn_type

--- a/vlib/v/checker/tests/method_ptr_used_as_argument_err.out
+++ b/vlib/v/checker/tests/method_ptr_used_as_argument_err.out
@@ -1,0 +1,14 @@
+vlib/v/checker/tests/method_ptr_used_as_argument_err.vv:8:16: error: cannot use `fn () int` as `int` in argument 1 to `add`
+    6 |     arg1 := '10'
+    7 |     arg2 := '20'
+    8 |     result := add(arg1.int, arg2.int)
+      |                   ~~~~~~~~
+    9 |     println(result)
+   10 | }
+vlib/v/checker/tests/method_ptr_used_as_argument_err.vv:8:26: error: cannot use `fn () int` as `int` in argument 2 to `add`
+    6 |     arg1 := '10'
+    7 |     arg2 := '20'
+    8 |     result := add(arg1.int, arg2.int)
+      |                             ~~~~~~~~
+    9 |     println(result)
+   10 | }

--- a/vlib/v/checker/tests/method_ptr_used_as_argument_err.vv
+++ b/vlib/v/checker/tests/method_ptr_used_as_argument_err.vv
@@ -1,0 +1,10 @@
+fn add(x int, y int) int {
+	return x + y
+}
+
+fn main() {
+	arg1 := '10'
+	arg2 := '20'
+	result := add(arg1.int, arg2.int)
+	println(result)
+}

